app/models/server_puppet_data.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+class ServerPuppetData
+  attr_reader :client
+
+  def initialize
+    @client = PuppetdbClient.new(Settings.PUPPETDB_ENDPOINT)
+  end
+
+  def self.run
+    new.load_data
+  end
+
+  def load_data
+    Server.find_each do |server|
```

Comment:
For efficiency now, yes. I can look into testing out various `:batch_size` values to really see if/how performance differs.